src/bls/bls.cpp

```diff
@@ -152,7 +162,14 @@ CBLSPublicKey CBLSPublicKey::AggregateInsecure(const std::vector<CBLSPublicKey>&
     }
 
     CBLSPublicKey ret;
-    ret.impl = Scheme(bls::bls_legacy_scheme.load())->Aggregate(vecPublicKeys);
+    try {
+        ret.impl = Scheme(bls::bls_legacy_scheme.load())->Aggregate(vecPublicKeys);
+    } catch (...) {
+        ret.fValid = false;
+        ret.cachedHash.SetNull();
+        return ret;
+    }
+
     ret.fValid = true;
```

Comment:
why not
```suggestion
        ret.fValid = true;
    } catch (...) {
        ret.fValid = false;
    }

```
? same below